### PR TITLE
OverTheBox - migration summary

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-593
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

OverTheBox - migration offer - resume view

### CODE REVIEW

File: packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -14,5 +14,26 @@
...
+  "overthebox_migration_resume_table_price_per_month": "Prix HT/mois",
+  "overthebox_migration_resume_table_offer": "Offre {{offerName}}",
+  "overthebox_migration_resume_table_hardware": "Boîtier",
+  "overthebox_migration_resume_contracts_title": "Contrats",
+  "overthebox_migration_resume_price_explain": "Votre nouvel abonnement mensuel s'élève à : {{price}} HT/mois, applicable dès la livraison de votre nouveau service. Vous en serez informé par e-mail.",
+  "overthebox_migration_resume_hardware_price": "Le prix du matériel inclus les frais de livraison de celui-ci.",
+  "overthebox_migration_resume_first_mensuality": "Votre première mensualité sera donc de {{price}} HT.",
+  "overthebox_migration_resume_check_label": "En cliquant sur \"Valider le changement d'offre\", vous reconnaissez avoir pris connaissance et acceptez les contrats suivants :",
+  "overthebox_migration_resume_apply_change_offer": "Valider le changement d'offre",
+  "overthebox_migration_resume_get_offer_detail_error": "Une erreur s'est produite lors de la récupération du détail de l'offre. {{error}}",
```
Comment (antonymarion) [inline packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json:32]: Same as previous one, do we put : {{ errorMsg }} or . {{errorMsg }}

File: packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -14,5 +14,26 @@
...
+  "overthebox_migration_resume_total": "Soit un total de :",
+  "overthebox_migration_resume_command_order_explain": "Pour finaliser la migration de votre offre, vous devez payer ",
+  "overthebox_migration_resume_command_order_help": "En savoir plus sur le règlement d'un bon de commande.",
+  "overthebox_migration_resume_command_order_link": "le bon de commande.",
+  "overthebox_migration_resume_command_order_link_id": "Bon de commande :",
+  "overthebox_migration_resume_launch_migration_error": "Une erreur s'est produite lors de la migration de l'offre. {{error}}"
```
Comment (antonymarion) [inline packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json:38]: Same as previous one, do we put : {{ errorMsg }} or . {{errorMsg }}

**Review by antonymarion — APPROVED** [review]

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+export default class OverTheBoxMigrationResumeService {
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js:1]: I think is summary (Synthèse in french) and not resume

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const params = {
+      offer,
+      hardware,
+    };
+    if (shippingContactID) {
+      params.shippingContactID = shippingContactID;
+    }
+    return this.$http.get(`/order/overTheBox/${serviceName}/migrate`, {
+      params,
+    });
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js:17]:
```suggestion
    this.$http.get(`/order/overTheBox/${serviceName}/migrate`, {
        params: {
          offer,
          hardware,
          ...(shippingContactID && { shippingContactID }),
        },
      });
```

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    if (shippingContactID) {
+      params.shippingContactID = shippingContactID;
+    }
+    if (hardwareName) {
+      params.hardwareName = hardwareName;
+    }
+    return this.$http.post(
+      `/overTheBox/${serviceName}/migration/changeOffers`,
+      params,
+    );
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.service.js:34]:
```suggestion
    return this.$http.post(
      `/overTheBox/${serviceName}/migration/changeOffers`,
      { params: {
      offer,
      shippingContactID,
      ...(hardwareName && { hardwareName }),
    }});
```

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.html [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+                    data-translate="overthebox_migration_resume_hardware_price"
+                ></p>
+
+                <p
+                    data-ng-if="$ctrl.offer.selectedHardware !== $ctrl.NO_HARDWARE"
+                    data-translate="overthebox_migration_resume_first_mensuality"
+                    data-translate-values="{price: $ctrl.firstMensuality.text}"
+                    class="font-weight-bold"
+                ></p>
+            </div>
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.html:120]:
```suggestion
                <p
                    data-translate="overthebox_migration_resume_price_explain"
                    data-translate-values="{price: $ctrl.offer.price.text}"
                ></p>

                <p
                    data-ng-if="$ctrl.offer.selectedHardware !== $ctrl.NO_HARDWARE"
                    data-translate="overthebox_migration_resume_hardware_price"
                ></p>

                <p
                    data-ng-if="$ctrl.offer.selectedHardware !== $ctrl.NO_HARDWARE"
                    data-translate="overthebox_migration_resume_first_mensuality"
                    data-translate-values="{price: $ctrl.firstMensuality.text}"
                    class="font-weight-bold"
                ></p>
```

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        // Display error message
+        const msg = error?.data?.message;
+        this.TucToast.error(
+          this.$translate.instant(
+            'overthebox_migration_resume_launch_migration_error',
+            {
+              errorMsg: msg,
+            },
+          ),
+        );
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js:100]:
```suggestion
        this.TucToast.error(
          this.$translate.instant(
            'overthebox_migration_resume_launch_migration_error',
            {
              errorMsg: error?.data?.message,
            },
          ),
        );
```

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  getFirstMonthPrice() {
+    if (this.offer.selectedHardware !== NO_HARDWARE) {
+      const value =
+        parseFloat(this.offer.price.value) +
+        parseFloat(this.offer.hardwarePrice.value);
+      this.firstMensuality = {
+        value,
+        currencyCode: 'EUR',
```
Comment (ghyenne) [inline packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js:40]: Shouldn't hardcoded, you need take the customer currency

File: packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json [hunk 9]
```diff
@@ -14,5 +14,26 @@
...
+  "overthebox_migration_resume_price_explain": "Votre nouvel abonnement mensuel s'élève à : {{price}} HT/mois, applicable dès la livraison de votre nouveau service. Vous en serez informé par e-mail.",
+  "overthebox_migration_resume_hardware_price": "Le prix du matériel inclus les frais de livraison de celui-ci.",
+  "overthebox_migration_resume_first_mensuality": "Votre première mensualité sera donc de {{price}} HT.",
+  "overthebox_migration_resume_check_label": "En cliquant sur \"Valider le changement d'offre\", vous reconnaissez avoir pris connaissance et acceptez les contrats suivants :",
+  "overthebox_migration_resume_apply_change_offer": "Valider le changement d'offre",
+  "overthebox_migration_resume_get_offer_detail_error": "Une erreur s'est produite lors de la récupération du détail de l'offre. {{error}}",
+  "overthebox_migration_resume_total": "Soit un total de :",
+  "overthebox_migration_resume_command_order_explain": "Pour finaliser la migration de votre offre, vous devez payer ",
+  "overthebox_migration_resume_command_order_help": "En savoir plus sur le règlement d'un bon de commande.",
+  "overthebox_migration_resume_command_order_link": "le bon de commande.",
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/migration/translations/Messages_fr_FR.json:36]: What would you suggest for this case?

File: packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js (hunk 7 above)
Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/migration/resume/overTheBox-migration-resume.controller.js:100]: No it doesn't work, it generated an errror with the $translate, that's why I did this.

**Review by ghyenne — APPROVED** [review]

**Pull request merged**